lyrics-core/src/main/java/com/flipkart/lyrics/processor/constructors/NoArgsConstructorHandler.java

```diff
@@ -36,9 +36,11 @@ public NoArgsConstructorHandler(Tune tune, MetaInfo metaInfo, RuleSet ruleSet) {
 
     @Override
     public void process(TypeSpec.Builder typeSpec, TypeModel typeModel) {
-        MethodSpec constructor = MethodSpec.constructorBuilder()
-                .addModifiers(Modifier.PUBLIC)
-                .build();
-        typeSpec.addMethod(constructor);
+        if (tune.isNoArgsConstructorNeeded()) {
```

Comment:
I always prefer a short-circuit, check for negative condition and return. More readable